Src/Amr/AMReX_StateData.cpp

```diff
@@ -527,9 +527,8 @@ StateData::FillBoundary (Box const&      bx,
                          int             num_comp)
 {
     BL_PROFILE("StateData::FillBoundary(geom)");
-    BL_ASSERT(bx.ixType() == desc->getType());
 
-    if (domain.contains(bx)) return;
+    if (domain.contains(enclosedCells(bx))) return;
```

Comment:
What if the domain is nodal type like Press_Type in IAMR?

Reply:
Good point. Although the function is passed a Geometry, and Geometry's domain is assumed to always be cell-centered (per comments in AMReX_Geometry.H, although this is never enforced or checked within Geometry), here, this is StateData's member variable domain, which can have any index type. ... In any case, I think my change should make this correct for any index type.